AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -658,8 +705,40 @@ - (void)dataControllerUnlockDataSource
   }
 }
 
-#pragma mark -
-#pragma mark ASRangeControllerDelegate.
+#pragma mark - ASCollectionViewDataControllerSource Supplementary view support
+
+- (ASDisplayNode *)dataController:(ASCollectionDataController *)dataController supplementaryNodeOfKind:(NSString *)kind atIndexPath:(NSIndexPath *)indexPath
+{
+  ASDisplayNode *node = [_asyncDataSource collectionView:self nodeForSupplementaryElementOfKind:kind atIndexPath:indexPath];
+  ASDisplayNodeAssert(node != nil, @"A node must be returned for a supplementary node");
+  ASDisplayNodeAssert(!node.nodeLoaded, @"The supplementary node must not be loaded");
```

Comment:
Is this assertion necessary?  I know that a loaded node would require main thread measurement (for now, before I can get to the thread affinity system), but ideally we could support it, in the event that the app has its own cache of nodes.  If there are a couple good reasons why this assertion must hold for now (e.g. nontrivial to support loaded nodes even in a future diff done by another contributor), go ahead and leave it.